src/VisualStudio/Core/Def/Implementation/Preview/FileChange.cs

```diff
@@ -225,14 +231,48 @@ internal override void GetDisplayData(VSTREEDISPLAYDATA[] pData)
             pData[0].Image = pData[0].SelectedImage = (ushort)StandardGlyphGroup.GlyphLibrary;
         }
 
-        public static IHierarchicalDifferenceCollection ComputeDiffSpans(SourceText oldText, SourceText newText, ITextDifferencingService diffService, CancellationToken cancellationToken)
+        private static IHierarchicalDifferenceCollection ComputeDiffSpans(ITextDifferencingService diffService, TextDocument left, TextDocument right, CancellationToken cancellationToken)
         {
+            // TODO: it would be nice to have a syntax based differ for presentation here, 
+            //       current way of just using text differ has its own issue, and using syntax differ in compiler that are for incremental parser
+            //       has its own drawbacks.
+
+            var oldText = left.GetTextAsync(cancellationToken).WaitAndGetResult(cancellationToken);
+            var newText = right.GetTextAsync(cancellationToken).WaitAndGetResult(cancellationToken);
+
+            // first try, cheapest way.
```

Comment:
this is best option I found for now. we probably want to have syntax based differ that are tweaked for good user presentation.

can't use compiler syntax differ since it will either return too small in some cases (in this particular case rather than "M1", it will say "1") or too much (compiler syntax differ will just report chunk of code as changed if it sees a lot of things in an area are changed since it doesn't matter for incremental parser).

always using "Word" level differ is too expansive as we learned from Dev12.

so, best bet probably is using TreeComparer in workspace layer once it gets to a form we can use here.